app/src/common/hooks/__tests__/useFhirResourceTreeData.test.ts

```diff
@@ -0,0 +1,62 @@
+import { structureDefinitionFactory } from "services/api/factory";
+
+import { buildTree, createElementNode } from "../useFhirResourceTreeData";
+
+const structureDefinition = structureDefinitionFactory.build();
+
+describe("BuildTree algorithm", () => {
+  it("Should create correct path", () => {
+    const elementDefinitions = structureDefinition.snapshot?.element ?? [];
+    const rootNode = createElementNode(elementDefinitions[0], {});
+    buildTree(elementDefinitions.slice(1), rootNode, rootNode);
+
+    //Basic tree generation
+    expect(rootNode.path).toBe(elementDefinitions[0].path);
+    expect(rootNode.children[0].path).toBe(elementDefinitions[1].path);
+
+    buildTree(
+      elementDefinitions,
+      rootNode,
+      rootNode.children[0],
+      rootNode.children[0].path
+    );
+  });
+
+  it("Should concat correctly the paths", () => {
+    const parentPath = "path.to.parent";
+    const elementDefinitions = structureDefinition.snapshot?.element ?? [];
+    const rootNode = createElementNode(elementDefinitions[0], {
+      parentPath,
+    });
+    buildTree(elementDefinitions.slice(1), rootNode, rootNode, rootNode.path);
+
+    //Basic tree generation
+    expect(rootNode.path).toBe(parentPath);
+    expect(rootNode.children[0].path).toBe(
+      `${parentPath}.${elementDefinitions[1].path
+        ?.split(".")
+        .slice(1)
+        .join(".")}`
```

Comment:
this is not always the same depending on the generated structuredefinition ? if it never changes, I think it's best to have a static string here

Reply:
ok I didn't see the factory yet, so this cannot be a static string, it's fine :+1: